Escape quotes in arguments for PowerShell

ChristianLW opened the pull request:
A fix for #318. This is just a simple patch and can of course be changed if necessary.

The `String.prototype.replaceAll.call(arg, ...)` instead of just `arg.replaceAll(...)` is because `appArguments` might contain things other than strings.

This doesn't add escaping to the app name itself.

I'm not sure if you want to set up a unit test for this scenario, and I'm not sure if this change marks a breaking API change (if some people already put <code>\`</code> in their strings even though that wouldn't work on other platforms).

ChristianLW committed:
Escape quotes in arguments for PowerShell

sindresorhus commented on index.js:
```diff
@@ -202,7 +202,7 @@ const baseOpen = async options => {
 
 		if (appArguments.length > 0) {
-			appArguments = appArguments.map(arg => `"\`"${arg}\`""`);
+			appArguments = appArguments.map(arg => `"\`"${String.prototype.replaceAll.call(arg, '"', '`"')}\`""`);
```
```suggestion
			appArguments = appArguments.map(arg => `"\`"${String(arg).replaceAll('"', '`"')}\`""`);
```

sindresorhus replied:
This does not handle the case where the quote is already escaped: `\"`

ChristianLW replied:
Sorry for the late response. Could you please elaborate on the `\"` escaping? I can't find anything on using `\` as an escape character in PowerShell.

Alternatively, would it make sense to use single quote strings, as they don't interpret special characters? You can include single quotes in single quote strings by doubling them. This would almost certainly eliminate any edge cases.

ChristianLW committed:
Use String() instead of String.prototype

First convert to string and then call replaceAll, instead of calling replaceAll from the string prototype

Co-authored-by: Sindre Sorhus <sindresorhus@gmail.com>
